match/pattern.md

```diff
@@ -24,11 +24,11 @@ x: c c: c
 x: 1
 ```
 
-在以上代码中，match作用域里的`x`这个变量被覆盖成了`'c'`，而出了这个作用域之后，变量又恢复成了原来的值。这和变量绑定的行为是一致的。
+在以上代码中，match作用域里的`x`这个变量被覆盖成了`'c'`，而出了这个作用域，变量又恢复成了原来的值。这和变量绑定的行为是一致的。
```

Comment:
我觉得这就是两个变量，被覆盖的是match作用域 **特定的名称的绑定** 而不是变量。
以下是 _The Rust Programming Language_ 的原文，用的都是绑定。

> There’s one pitfall with patterns: like anything that introduces a new binding, they introduce shadowing. For example:
> ```
> let x = 1;
> let c = 'c';

> match c {
    x => println!("x: {} c: {}", x, c),
}

> println!("x: {}", x)
> ```
This prints:

> ```
> x: c c: c 
> x: 1
> ```
> In other words, x => matches the pattern and introduces a new binding named x. This new binding is in scope for the match arm and takes on the value of c. Notice that the value of x outside the scope of the match has no bearing on the value of x within it. Because we already have a binding named x, this new x shadows it.